### ISSUE TITLE
Storage total inflow volume and evaporation loss percent is not reported correctly

### ISSUE THREAD

**Issue by shenh** [USEPA/Stormwater-Management-Model#97]
A test model was set up to check the evaporation volume. There is no inflow into the storage node, and the conduit inlet offset is set high enough to prevent storage outflow. 

Problem 1: in the Summary Results > Node Inflow > Total Inflow Volume, SU1 has a total inflow volume of 100 ML, which is the initial storage volume. 
Problem 2: in the Summary Results > Storage Volume > Evaporation Loss %, SU1 has a value of 5%. This does not match the Flow Routing Continuity values: 1 / (Initial Stored Volume 9.999 + External Inflow 0). 

The two problems may be related to massbal.c > massbal_open > for (j = 0; j < Nobjects[NODE]; j++) NodeInflow[j] = Node[j].newVolume.

The reported Infiltration Loss % may be incorrect for the same reason. 

Model input file ------
[TITLE]
;;Project Title/Notes

[OPTIONS]
;;Option             Value
FLOW_UNITS           CMS
INFILTRATION         HORTON
FLOW_ROUTING         DYNWAVE
LINK_OFFSETS         DEPTH
MIN_SLOPE            0
ALLOW_PONDING        NO
SKIP_STEADY_STATE    NO

START_DATE           11/16/2022
START_TIME           00:00:00
REPORT_START_DATE    11/16/2022
REPORT_START_TIME    00:00:00
END_DATE             11/17/2022
END_TIME             00:00:00
SWEEP_START          01/01
SWEEP_END            12/31
DRY_DAYS             0
REPORT_STEP          00:01:00
WET_STEP             00:01:00
DRY_STEP             00:01:00
ROUTING_STEP         5
RULE_STEP            00:00:00

INERTIAL_DAMPING     PARTIAL
NORMAL_FLOW_LIMITED  BOTH
FORCE_MAIN_EQUATION  H-W
VARIABLE_STEP        0.75
LENGTHENING_STEP     0
MIN_SURFAREA         0
MAX_TRIALS           8
HEAD_TOLERANCE       0.0015
SYS_FLOW_TOL         5
LAT_FLOW_TOL         5
MINIMUM_STEP         0.5
THREADS              12

[EVAPORATION]
;;Data Source    Parameters
;;-------------- ----------------
CONSTANT         1000
DRY_ONLY         NO

[OUTFALLS]
;;Name           Elevation  Type       Stage Data       Gated    Route To        
;;-------------- ---------- ---------- ---------------- -------- ----------------
OF1              0          FREE                        NO                       

[STORAGE]
;;Name           Elev.    MaxDepth   InitDepth  Shape       Curve Name/Params            SurDepth Fevap    Psi      Ksat     IMD     
;;-------------- -------- ---------- ---------- ----------- ---------------------------- -------- -------- -------- -------- --------
SU1              0        100        10         FUNCTIONAL  0         0         10000    0        1       

[CONDUITS]
;;Name           From Node        To Node          Length     Roughness  InOffset   OutOffset  InitFlow   MaxFlow   
;;-------------- ---------------- ---------------- ---------- ---------- ---------- ---------- ---------- ----------
C1               SU1              OF1              106.721    0.01       15         0          0          0         

[XSECTIONS]
;;Link           Shape        Geom1            Geom2      Geom3      Geom4      Barrels    Culvert   
;;-------------- ------------ ---------------- ---------- ---------- ---------- ---------- ----------
C1               CIRCULAR     1                0          0          0          1                    

[LOSSES]
;;Link           Kentry     Kexit      Kavg       Flap Gate  Seepage   
;;-------------- ---------- ---------- ---------- ---------- ----------

[REPORT]
;;Reporting Options
INPUT      YES
CONTROLS   NO
SUBCATCHMENTS ALL
NODES ALL
LINKS ALL

[TAGS]

[MAP]
DIMENSIONS       -334.600787012332 167.327927334673 -218.181472741038 239.471241555492
UNITS            Meters

[COORDINATES]
;;Node           X-Coord            Y-Coord           
;;-------------- ------------------ ------------------
OF1              -223.473           170.607           
SU1              -329.309           184.324           

[VERTICES]
;;Link           X-Coord            Y-Coord           
;;-------------- ------------------ ------------------

**Comment by dickinsonre** [comment]
You are using a very high evap rate = 1000 mm/day

  **************************        Volume        Volume
  Flow Routing Continuity        hectare-m      10^6 ltr
  **************************     ---------     ---------
  Dry Weather Inflow .......         0.000         0.000
  Wet Weather Inflow .......         0.000         0.000
  Groundwater Inflow .......         0.000         0.000
  RDII Inflow ..............         0.000         0.000
  External Inflow ..........         0.000         0.000
  External Outflow .........         0.000         0.000
  Flooding Loss ............         0.000         0.000
  Evaporation Loss .........         1.000        10.000
  Exfiltration Loss ........         0.000         0.000
  Initial Stored Volume ....         9.999        99.990
  Final Stored Volume ......         8.999        89.991
  Continuity Error (%) .....        -0.001

**Comment by dickinsonre** [comment]
If i multiply 10,000 square meters times 1000 mm/day for one day I do get 1 hectare-m of evaporation loss.

**Comment by shenh** [comment]
Thanks Bob for reviewing this. The system-wise summary is correct. The evaporate rate is set to a very high value so that the evaporation loss is visible within a short simulation duration.

The ultimate goal is to calculate storage node evaporation loss volume from report values at the storage. 

I think the issue is in NodeInflow[j]. It also includes the initial volume. 

Hailiang

![2022-11-17_17-09-15](https://user-images.githubusercontent.com/4484380/202570662-ef95eb8a-3d4b-400f-b323-ed98d26378b3.png)

**Comment by LRossman** [comment]
By definition the value in the Total Inflow Volume column of the Node Inflow Summary Results table includes any initial volume in the node so no action is warranted.

As for the % Evaporation Loss value in the Storage Volume table, aside from being ambiguous (% of what) it's not being calculated correctly:
```
 pctEvapLoss = StorageStats[k].evapLosses / addedVol * 100.0;
```
where
```
addedVol = NodeInflow[j] + StorageStats[k].initVol;
```
In the supplied example `addedVol `is double counting the initial storage volume (since it's already included in `NodeInflow`) producing a % Evaporation Loss about half of what it should be.

A less ambiguous definition of % Evaporation Loss would be Total Evaporation Volume / Average Storage Volume * 100. Using this definition in the code produces an Evaporation Loss of 11% for the example.

A similar definition should be used for % Exfiltration Loss.

**Comment by shenh** [comment]
Thanks Lew for clarifying the reported total inflow volume.

Redefining % Evaporation Loss to Total Evaporation Volume / Average Storage Volume * 100 may generate a value greater than 100%, which may cause confusion. But I agree it will be easier to calculate evaporation loss volume since Average Storage Volume is also reported. Perhaps % Evaporation Loss can be renamed to Evaporation Loss Ratio? So, a value greater than 1 will make sense. 

The same applies to % Exfiltration Loss.

**Comment by LRossman** [comment]
After thinking more about this I've come full circle and decided it's best to keep the original definition of % Evaporation Loss (percent of total storage volume that is evaporated) and merely correct its calculation:
```
pctEvapLoss = StorageStats[k].evapLosses /  NodeInflow[j] * 100.0;
```
As noted previously, `NodeInflow `includes the storage unit's initial volume plus the total volume of all inflow to it. This is the  amount of water available to be evaporated. Defining evaporation loss by this percentage gives the user a better idea of whether evaporation is significant or not which would be difficult to determine if only the total evaporated volume were reported. If the user needed to know what this volume was they can multiply the storage node's Total Inflow Volume value in the Node Inflow Summary table by the % evaporation loss.

**Comment by shenh** [comment]
Thanks Lew, this makes sense.

**Comment by michaeltryby** [comment]
Closed in Release v5.2.2
